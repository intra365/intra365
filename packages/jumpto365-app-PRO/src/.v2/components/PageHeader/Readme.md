



```js
<PageHeader title="Page Header" color="#2a7ab9" iconUrl="https://jumpto365.com/resources/images/app/Word.png" />
```

```js
<PageHeader title="Page Header no logo" color="#2a7ab9" xiconUrl="https://jumpto365.com/resources/images/app/Word.png" />
```