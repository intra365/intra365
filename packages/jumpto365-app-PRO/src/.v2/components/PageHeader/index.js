import React, { Component } from 'react';
import PropTypes from 'prop-types'
/**
 * Describe overall purpose of the component
 *
 * @export
 * @class PageHeader
 * @extends {Component}
 */
export default class PageHeader extends Component {

    static propTypes  = {
        title : PropTypes.string,
        color : PropTypes.string,
        icon: PropTypes.string,
    }
    

    constructor(props) {
        super(props);
    }

    /**
     * Required method return the output of the component
     *
     * @returns
     * @memberof PageHeader
     */
    render() {
        var icon = this.props.icon ? <img src={this.props.icon} style={{marginTop:"8px", 
        textAlign: 'center', marginRight:"4px",height:"32px",width:"auto"}}/> : null
        return (
            <div id="pageheader" style={{marginBottom:"8px"}}>
<div  style={{ clear:"both", backgroundColor : this.props.color, marginLeft: '-8px',
                marginRight: '-8px'}}>
            <div className="ms-Grid" style={{
                
                lineHeight: '48px',
                
                height: '48px',
                color:"#ffffff",
               
                
            }}>
            <div className="ms-Grid-row" >
                <div className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight" style={{marginLeft:"8px",maxHeight:"48px",overflow:"clip"}}>
                <span style={{display:"inline-block"}}>{icon}</span> <span  title={this.props.title} style={{display:"inline", maxHeight:"48px", verticalAlign:"top",width:"100%",    overflow: 'hidden',
              textOverflow: 'ellipsis',
              whiteSpace: 'inherit'}}> 
                
                {this.props.title}</span>
                </div>
                <div className="ms-Grid-col ms-sm2  " style={{textAlign:"right"}} >
                {this.props.children}
                </div>
            </div>
            
        </div>
        <div style={{ clear:"both"}} />

        </div>
</div>
        )
    }
}

