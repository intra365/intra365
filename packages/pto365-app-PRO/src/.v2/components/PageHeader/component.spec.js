import React, { Component } from 'react';

import PageHeader from '.'

test('Check that it renders', () => {

    const wrapper = shallow(<PageHeader about="🍕"/>);

    expect(wrapper).toMatchSnapshot();
 
}

)