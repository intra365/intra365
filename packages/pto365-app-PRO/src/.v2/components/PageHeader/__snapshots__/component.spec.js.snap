// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader
    about="🍕"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "backgroundColor": undefined,
                        "clear": "both",
                        "marginLeft": "-8px",
                        "marginRight": "-8px",
                      }
        }
>
        <div
                className="ms-Grid"
                style={
                        Object {
                                "color": "#ffffff",
                                "height": "48px",
                                "lineHeight": "48px",
                              }
                }
        >
                <div
                        className="ms-Grid-row"
                >
                        <div
                                className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                                style={
                                        Object {
                                                "marginLeft": "8px",
                                                "maxHeight": "48px",
                                                "overflow": "clip",
                                              }
                                }
                        >
                                <span
                                        style={
                                                Object {
                                                        "display": "inline-block",
                                                      }
                                        }
                                />
                                 
                                <span
                                        style={
                                                Object {
                                                        "display": "inline",
                                                        "maxHeight": "48px",
                                                        "textOverflow": "clip",
                                                        "verticalAlign": "top",
                                                        "width": "100%",
                                                      }
                                        }
                                        title={undefined}
                                />
                        </div>
                        <div
                                className="ms-Grid-col ms-sm2  "
                                style={
                                        Object {
                                                "textAlign": "right",
                                              }
                                }
                        />
                </div>
        </div>
        <div
                style={
                        Object {
                                "clear": "both",
                              }
                }
        />
</div>,
      "id": "pageheader",
      "style": Object {
        "marginBottom": "8px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ms-Grid"
            style={
                        Object {
                                    "color": "#ffffff",
                                    "height": "48px",
                                    "lineHeight": "48px",
                                  }
            }
>
            <div
                        className="ms-Grid-row"
            >
                        <div
                                    className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                                    style={
                                                Object {
                                                            "marginLeft": "8px",
                                                            "maxHeight": "48px",
                                                            "overflow": "clip",
                                                          }
                                    }
                        >
                                    <span
                                                style={
                                                            Object {
                                                                        "display": "inline-block",
                                                                      }
                                                }
                                    />
                                     
                                    <span
                                                style={
                                                            Object {
                                                                        "display": "inline",
                                                                        "maxHeight": "48px",
                                                                        "textOverflow": "clip",
                                                                        "verticalAlign": "top",
                                                                        "width": "100%",
                                                                      }
                                                }
                                                title={undefined}
                                    />
                        </div>
                        <div
                                    className="ms-Grid-col ms-sm2  "
                                    style={
                                                Object {
                                                            "textAlign": "right",
                                                          }
                                    }
                        />
            </div>
</div>,
          <div
            style={
                        Object {
                                    "clear": "both",
                                  }
            }
/>,
        ],
        "style": Object {
          "backgroundColor": undefined,
          "clear": "both",
          "marginLeft": "-8px",
          "marginRight": "-8px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="ms-Grid-row"
>
              <div
                            className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                            style={
                                          Object {
                                                        "marginLeft": "8px",
                                                        "maxHeight": "48px",
                                                        "overflow": "clip",
                                                      }
                            }
              >
                            <span
                                          style={
                                                        Object {
                                                                      "display": "inline-block",
                                                                    }
                                          }
                            />
                             
                            <span
                                          style={
                                                        Object {
                                                                      "display": "inline",
                                                                      "maxHeight": "48px",
                                                                      "textOverflow": "clip",
                                                                      "verticalAlign": "top",
                                                                      "width": "100%",
                                                                    }
                                          }
                                          title={undefined}
                            />
              </div>
              <div
                            className="ms-Grid-col ms-sm2  "
                            style={
                                          Object {
                                                        "textAlign": "right",
                                                      }
                            }
              />
</div>,
            "className": "ms-Grid",
            "style": Object {
              "color": "#ffffff",
              "height": "48px",
              "lineHeight": "48px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                  style={
                                    Object {
                                                      "marginLeft": "8px",
                                                      "maxHeight": "48px",
                                                      "overflow": "clip",
                                                    }
                  }
>
                  <span
                                    style={
                                                      Object {
                                                                        "display": "inline-block",
                                                                      }
                                    }
                  />
                   
                  <span
                                    style={
                                                      Object {
                                                                        "display": "inline",
                                                                        "maxHeight": "48px",
                                                                        "textOverflow": "clip",
                                                                        "verticalAlign": "top",
                                                                        "width": "100%",
                                                                      }
                                    }
                                    title={undefined}
                  />
</div>,
                <div
                  className="ms-Grid-col ms-sm2  "
                  style={
                                    Object {
                                                      "textAlign": "right",
                                                    }
                  }
/>,
              ],
              "className": "ms-Grid-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      style={
                                            Object {
                                                                  "display": "inline-block",
                                                                }
                      }
/>,
                    " ",
                    <span
                      style={
                                            Object {
                                                                  "display": "inline",
                                                                  "maxHeight": "48px",
                                                                  "textOverflow": "clip",
                                                                  "verticalAlign": "top",
                                                                  "width": "100%",
                                                                }
                      }
                      title={undefined}
/>,
                  ],
                  "className": "ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight",
                  "style": Object {
                    "marginLeft": "8px",
                    "maxHeight": "48px",
                    "overflow": "clip",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                      "style": Object {
                        "display": "inline-block",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "style": Object {
                        "display": "inline",
                        "maxHeight": "48px",
                        "textOverflow": "clip",
                        "verticalAlign": "top",
                        "width": "100%",
                      },
                      "title": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "ms-Grid-col ms-sm2  ",
                  "style": Object {
                    "textAlign": "right",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "style": Object {
              "clear": "both",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "backgroundColor": undefined,
                              "clear": "both",
                              "marginLeft": "-8px",
                              "marginRight": "-8px",
                            }
          }
>
          <div
                    className="ms-Grid"
                    style={
                              Object {
                                        "color": "#ffffff",
                                        "height": "48px",
                                        "lineHeight": "48px",
                                      }
                    }
          >
                    <div
                              className="ms-Grid-row"
                    >
                              <div
                                        className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                                        style={
                                                  Object {
                                                            "marginLeft": "8px",
                                                            "maxHeight": "48px",
                                                            "overflow": "clip",
                                                          }
                                        }
                              >
                                        <span
                                                  style={
                                                            Object {
                                                                      "display": "inline-block",
                                                                    }
                                                  }
                                        />
                                         
                                        <span
                                                  style={
                                                            Object {
                                                                      "display": "inline",
                                                                      "maxHeight": "48px",
                                                                      "textOverflow": "clip",
                                                                      "verticalAlign": "top",
                                                                      "width": "100%",
                                                                    }
                                                  }
                                                  title={undefined}
                                        />
                              </div>
                              <div
                                        className="ms-Grid-col ms-sm2  "
                                        style={
                                                  Object {
                                                            "textAlign": "right",
                                                          }
                                        }
                              />
                    </div>
          </div>
          <div
                    style={
                              Object {
                                        "clear": "both",
                                      }
                    }
          />
</div>,
        "id": "pageheader",
        "style": Object {
          "marginBottom": "8px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ms-Grid"
              style={
                            Object {
                                          "color": "#ffffff",
                                          "height": "48px",
                                          "lineHeight": "48px",
                                        }
              }
>
              <div
                            className="ms-Grid-row"
              >
                            <div
                                          className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                                          style={
                                                        Object {
                                                                      "marginLeft": "8px",
                                                                      "maxHeight": "48px",
                                                                      "overflow": "clip",
                                                                    }
                                          }
                            >
                                          <span
                                                        style={
                                                                      Object {
                                                                                    "display": "inline-block",
                                                                                  }
                                                        }
                                          />
                                           
                                          <span
                                                        style={
                                                                      Object {
                                                                                    "display": "inline",
                                                                                    "maxHeight": "48px",
                                                                                    "textOverflow": "clip",
                                                                                    "verticalAlign": "top",
                                                                                    "width": "100%",
                                                                                  }
                                                        }
                                                        title={undefined}
                                          />
                            </div>
                            <div
                                          className="ms-Grid-col ms-sm2  "
                                          style={
                                                        Object {
                                                                      "textAlign": "right",
                                                                    }
                                          }
                            />
              </div>
</div>,
            <div
              style={
                            Object {
                                          "clear": "both",
                                        }
              }
/>,
          ],
          "style": Object {
            "backgroundColor": undefined,
            "clear": "both",
            "marginLeft": "-8px",
            "marginRight": "-8px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="ms-Grid-row"
>
                <div
                                className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                                style={
                                                Object {
                                                                "marginLeft": "8px",
                                                                "maxHeight": "48px",
                                                                "overflow": "clip",
                                                              }
                                }
                >
                                <span
                                                style={
                                                                Object {
                                                                                "display": "inline-block",
                                                                              }
                                                }
                                />
                                 
                                <span
                                                style={
                                                                Object {
                                                                                "display": "inline",
                                                                                "maxHeight": "48px",
                                                                                "textOverflow": "clip",
                                                                                "verticalAlign": "top",
                                                                                "width": "100%",
                                                                              }
                                                }
                                                title={undefined}
                                />
                </div>
                <div
                                className="ms-Grid-col ms-sm2  "
                                style={
                                                Object {
                                                                "textAlign": "right",
                                                              }
                                }
                />
</div>,
              "className": "ms-Grid",
              "style": Object {
                "color": "#ffffff",
                "height": "48px",
                "lineHeight": "48px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight"
                    style={
                                        Object {
                                                            "marginLeft": "8px",
                                                            "maxHeight": "48px",
                                                            "overflow": "clip",
                                                          }
                    }
>
                    <span
                                        style={
                                                            Object {
                                                                                "display": "inline-block",
                                                                              }
                                        }
                    />
                     
                    <span
                                        style={
                                                            Object {
                                                                                "display": "inline",
                                                                                "maxHeight": "48px",
                                                                                "textOverflow": "clip",
                                                                                "verticalAlign": "top",
                                                                                "width": "100%",
                                                                              }
                                        }
                                        title={undefined}
                    />
</div>,
                  <div
                    className="ms-Grid-col ms-sm2  "
                    style={
                                        Object {
                                                            "textAlign": "right",
                                                          }
                    }
/>,
                ],
                "className": "ms-Grid-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        style={
                                                Object {
                                                                        "display": "inline-block",
                                                                      }
                        }
/>,
                      " ",
                      <span
                        style={
                                                Object {
                                                                        "display": "inline",
                                                                        "maxHeight": "48px",
                                                                        "textOverflow": "clip",
                                                                        "verticalAlign": "top",
                                                                        "width": "100%",
                                                                      }
                        }
                        title={undefined}
/>,
                    ],
                    "className": "ms-Grid-col ms-sm9 ms-fontSize-xxl ms-fontWeight-semilight",
                    "style": Object {
                      "marginLeft": "8px",
                      "maxHeight": "48px",
                      "overflow": "clip",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": null,
                        "style": Object {
                          "display": "inline-block",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "style": Object {
                          "display": "inline",
                          "maxHeight": "48px",
                          "textOverflow": "clip",
                          "verticalAlign": "top",
                          "width": "100%",
                        },
                        "title": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "ms-Grid-col ms-sm2  ",
                    "style": Object {
                      "textAlign": "right",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "style": Object {
                "clear": "both",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
